src/main/java/io/nats/client/support/Validator.java

```diff
@@ -203,12 +203,13 @@ public static boolean containsDotWildGt(String s) {
         return false;
     }
 
-    public static boolean containsWildGt(String s) {
+    public static boolean containsWildGtDollar(String s) {
         if (s != null) {
             for (int i = 0; i < s.length(); i++) {
                 switch (s.charAt(i)) {
                     case '*':
                     case '>':
+                    case '$':
```

Comment:
Should check for space and tab too (had to double-check myself):

https://docs.nats.io/nats-protocol/nats-protocol#protocol-conventions

Reply:
fixed